cli/cli_app.go

```diff
@@ -264,6 +266,57 @@ func shouldPrintTerraformHelp(terragruntOptions *options.TerragruntOptions) bool
 	return false
 }
 
+// loadEnvVars loads environment variable definitions from a shell command.
+// Currently the function expects KEY=value output to STDOUT
+// the method returns the old evironment variables map (dictionary) if the
+// changes should be transient.
+func loadEnvVars(terragruntOptions *options.TerragruntOptions, curHook config.Hook) (map[string]string, error) {
+	var oldEnvMap map[string]string
+
+	if curHook.LoadEnvVars != nil {
+		terragruntOptions.Logger.Printf("Loading environment for hook: %s", curHook.Name)
+
+		actionToExecute := curHook.LoadEnvVars.Execute[0]
+		actionParams := curHook.LoadEnvVars.Execute[1:]
+		out, err := shell.RunShellCommandWithOutput(terragruntOptions, "", actionToExecute, actionParams...)
+		if err != nil {
+			terragruntOptions.Logger.Printf("Error running hook %s while loading environment variables with message: %s", curHook.Name, err.Error())
+			return nil, err
+		}
+		// parse out from script to map
+		m := make(map[string]string)
+		scanner := bufio.NewScanner(strings.NewReader(out.Stdout))
+		for scanner.Scan() {
+			txt := scanner.Text()
+			idx := strings.Index(txt, "=")
+			if idx > 0 {
+				key := strings.TrimSpace(txt[0:idx])
+				value := txt[idx+1 : len(txt)]
+				m[key] = value
+			} else {
+				terragruntOptions.Logger.Print("Invalid key,value pair")
+			}
+		}
+		targetEnvMap := terragruntOptions.Env
+		if curHook.LoadEnvVars.Transient {
+			oldEnvMap = terragruntOptions.Env
```

Comment:
Will this preserve the vars correctly if you overwrite `terragruntOptions.Env` later?